skorch/net.py

```diff
@@ -1421,21 +1409,33 @@ def save_params(
         if f_history is not None:
             self.history.to_file(f_history)
 
-    def _get_state_dict(self, f):
-        """Load torch object (module, optimizer) using desired device."""
-        map_location = torch.device(self.device)
+    def _get_map_location(self, target_device):
+        """Determine the location to map loaded data (e.g., weights)
+        for a given target device (e.g. 'cuda').
+        """
+        map_location = torch.device(target_device)
 
         # The user wants to use CUDA but there is no CUDA device
         # available, thus fall back to CPU.
-        if self.device.startswith('cuda') and not torch.cuda.is_available():
+        if target_device.startswith('cuda') and not torch.cuda.is_available():
             warnings.warn(
-                "Model configured to use CUDA but no CUDA devices "
-                "available. Loading on CPU instead.",
-                ResourceWarning)
-            self.device = 'cpu'
+                'Requested to load data to CUDA but no CUDA devices '
+                'are available. Loading on CPU instead.',
+                DeviceWarning)
             map_location = torch.device('cpu')
+        return map_location
 
-        return torch.load(f, map_location=map_location)
+    def _handle_mismatched_devices(self, net_state, map_device):
+        """Warn the user and update the network state to use a non-cuda device
+        in case the network state requests e.g. CUDA but CUDA is not available.
+        """
+        requested_device = net_state['device']
+        if torch.device(requested_device) != torch.device(map_device):
```

Comment:
Is it necessary to use `torch.device`?

```suggestion
        if requested_device != map_device:
```

Maybe `torch.device` covers more but I wouldn't know what (e.g. even with 1 GPU, `torch.device('cuda:0') != torch.device('cuda')`)

Reply:
It handles the case where one of the values is a string instead of a device:

```
device(device('cuda')) == device('cuda')
```

The case of explicitly enumerated devices could be tricky, e.g. when switching from a single gpu machine to a multi gpu machine where it is now necessary to specify a device qualifier. We should probably resort to device type prefix matching to avoid this issue (instead of a equality test).